smithy-model/src/main/java/software/amazon/smithy/model/Model.java

```diff
@@ -192,6 +213,424 @@ private TraitCache getTraitCache() {
         return Collections.unmodifiableSet(mappings.getOrDefault(trait, Collections.emptySet()));
     }
 
+    /**
+     * Gets an immutable set of all bigDecimals in the Model.
+     *
+     * @return Returns the Set of {@code bigDecimals}s.
+     */
+    public Set<BigDecimalShape> getBigDecimalShapes() {
+        return toSet(BigDecimalShape.class);
+    }
+
+    /**
+     * Gets an immutable set of all bigDecimals in the Model that have a specific trait.
+     *
+     * @param trait The exact trait class to look for on shapes.
+     * @return Returns the set of {@code bigdecimal}s that have a specific trait.
+     */
+    public Set<BigDecimalShape> getBigDecimalShapesWithTrait(Class<? extends Trait> trait) {
+        return new ShapeTypeFilteredSet<>(getShapesWithTrait(trait), BigDecimalShape.class);
+    }
```

Comment:
I'm not really a huge fan of having methods for each shape type, but this would head-off fat-fingering and it's a bit easier to understand for people less experienced with java.

Reply:
This is also much more usable than having two `Class` params generally.